Assignment2 - Sergei Ashikhmin & Timothy Nelson

SergeiAshikhmin opened the pull request:
Sergei - @SergeiAshikhmin 
Timothy - @timmytaters 

Azure link: https://lemon-flower-0c0aec71e.5.azurestaticapps.net/

SergeiAshikhmin committed:
ci: add Azure Static Web Apps workflow file
on-behalf-of: @Azure opensource@microsoft.com

SergeiAshikhmin committed:
Update azure-static-web-apps-thankful-plant-050d9af1e.yml

SergeiAshikhmin committed:
Update azure-static-web-apps-thankful-plant-050d9af1e.yml

SergeiAshikhmin committed:
Fix: game file path

SergeiAshikhmin committed:
Merge branch 'main' of https://github.com/SergeiAshikhmin/EWU-CSCD379-2024-Spring

SergeiAshikhmin committed:
Add vitest package

SergeiAshikhmin committed:
Add more packages

SergeiAshikhmin committed:
Update azure-static-web-apps-thankful-plant-050d9af1e.yml

SergeiAshikhmin committed:
Bunch of packages and changes

SergeiAshikhmin committed:
Merge branch 'main' of https://github.com/SergeiAshikhmin/EWU-CSCD379-2024-Spring

SergeiAshikhmin committed:
Initialize project

SergeiAshikhmin committed:
Add Vuerify3

SergeiAshikhmin committed:
Complete Vuetify3 set up

SergeiAshikhmin committed:
Add Vitest and dummy test file

SergeiAshikhmin committed:
Add two pages, some logic to game, and more tests

SergeiAshikhmin committed:
ci: add Azure Static Web Apps workflow file
on-behalf-of: @Azure opensource@microsoft.com

SergeiAshikhmin committed:
Update azure-static-web-apps-lemon-flower-0c0aec71e.yml

SergeiAshikhmin committed:
Update azure-static-web-apps-lemon-flower-0c0aec71e.yml

SergeiAshikhmin committed:
Add azure link

Commit:
Some cleaning

SergeiAshikhmin committed:
add more tests

Commit:
Removed wordle-web and renamed

Commit:
Reverted to wordle

we misunderstood the assignment lol

SergeiAshikhmin committed:
Update azure-static-web-apps-lemon-flower-0c0aec71e.yml

SergeiAshikhmin committed:
Update azure-static-web-apps-lemon-flower-0c0aec71e.yml

SergeiAshikhmin committed:
Feature: task #2 add Name of the app and icons

SergeiAshikhmin committed:
Feature: task #2 add hamburger nav and about page

Commit:
Caught up to class

Commit:
Reverted Sergei's about menu

Commit:
Made Icon and title clickable, removed home and

Commit:
Finishing 2-4

Commit:
Error in tests, will replace

SergeiAshikhmin committed:
Add more text to About page

Commit:
WordList start

Opens and closes at bottom, and clicking each word types it in, but doesn't have updating valid words

Commit:
Small adjustments to wordlist presentation

Commit:
Added drop shadow

Commit:
Added hover to wordlist

Commit:
Didn't realize modes were variants of dark/light

For some reason the dark mode ones dont work

Commit:
Theme fixes

SergeiAshikhmin committed:
Update README.md

SergeiAshikhmin committed:
Add Settings dialog and more themes

SergeiAshikhmin committed:
Feature: Task #3 complete

SergeiAshikhmin committed:
Merge branch 'Assignment2' of https://github.com/SergeiAshikhmin/EWU-CSCD379-2024-Spring into Assignment2

SergeiAshikhmin committed:
Fix: add close btn to settings

SergeiAshikhmin committed:
Fix: more styling for themes settings

SergeiAshikhmin committed:
Fix: shadows and responsiveness

SergeiAshikhmin committed:
Fix: disable-resize-watcher

HadiM2 commented:
Reviewing

HadiM2 commented on Assignment2/tests/game.test.ts:
make sure to add a test for valid words method

SergeiAshikhmin replied:
Will do :)

HadiM2 commented on Assignment2/scripts/word.ts:
✅

HadiM2 commented on Assignment2/pages/about.vue:
✅

HadiM2 commented on Assignment2/components/SettingsDialog.vue:
✅

HadiM2 reviewed (commented):
Create a custom component to show available words ✅

Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ✅
Create unit tests for the validWords method ❌
Display the number of valid words based on the entered letters ❌
When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
When a word is clicked, the word should be entered as the guess but not submitted ✅
Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅

Convert the sidebar to an App Bar ✅

Name of the app on the left with an icon ✅
Choose an icon for the app (from Material Design or somewhere on the web) ✅
Clicking the name and icon of the app take you to the home page ✅
Hamburger menu on the right ✅
Menu option for an About page ✅
    Create an About page that has a few sentences about this class project and how great Meg is ✅
    Make sure this has a back button ✅

Add a settings dialog ✅

Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
Option for dark and light mode ✅
Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
The above two items must be implemented with built in Veutify features ✅

Style the main game page ✅

Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
Add gradient to guesses blocks and keys (with CSS) ❌
Add drop shadow to keys (No CSS) ❌
Make the app responsive (No CSS) ✅

Include a link to your Azure App in your PR comments. ✅

Extra Credit

Add an appropriate sound when the UI letter buttons are clicked (5 points) ❌
Add creative styling following good UI/UX practices ❌
    Look at products by Google or some of the vuetify components for good UX/UI ideas
    https://m2.material.io/design/guidelines-over

SergeiAshikhmin committed:
Working on Task 3

SergeiAshikhmin committed:
Code cleanup

MattA41 commented:
reviewing

MattA41 commented on Assignment2/components/SettingsDialog.vue:
```diff
@@ -0,0 +1,106 @@
+<template>
```
I like your use of view components to allow for someone to select themes

SergeiAshikhmin replied:
Thank you :)

MattA41 commented on Assignment2/pages/index.vue:
```diff
@@ -0,0 +1,5853 @@
+}
+
+// From: https://github.com/charlesreid1/five-letter-words/blob/master/sgb-words.txt
+var WordList = [
```
You do not have to write out the whole word list you can import it from the .ts document. Having it somewhere else would make it more readable

SergeiAshikhmin replied:
Yeah, I know. This feature was in development and we didn't care about clean code at that moment. But thanks for the catch. 😄

MattA41 commented on Assignment2/plugins/vuetify.ts:
```diff
@@ -0,0 +1,187 @@
+import "vuetify/styles";
+import { createVuetify } from "vuetify";
+
+export default defineNuxtPlugin((app) => {
```
I like the themes you made

SergeiAshikhmin replied:
Thank you

MattA41 reviewed (commented):
Over view: It looks good, you guys did some stuff differently than I did but that does not mean its wrong. 

EWU-CSCD378-2024-Spring

## Assignment 2

The purpose of this assignment is to solidify your learning of:
- Front end component development
- Using Vuetify's components
- Building a new custom component
- Creating new pages

## Instructions
Note: Custom CSS can only be used where specified.

1. Create a custom component to show available words ✅
- Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ❌
- Create unit tests for the validWords method ❌
- Display the number of valid words based on the entered letters ❌
- When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
- When a word is clicked, the word should be entered as the guess but not submitted ✅
- Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅

2. Convert the sidebar to an App Bar ✅
- Name of the app on the left with an icon ✅
- Choose an icon for the app (from Material Design or somewhere on the web) ✅
- Clicking the name and icon of the app take you to the home page ✅
- Hamburger menu on the right ✅
- Menu option for an About page ✅
  - Create an About page that has a few sentences about this class project and how great Meg is ✅
  - Make sure this has a back button ✅

3. Add a settings dialog ✅
- Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
- Option for dark and light mode ✅
- Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
- The above two items must be implemented with built in Veutify features ✅

4. Style the main game page ✅
- Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
- Add gradient to guesses blocks and keys (with CSS) ✅
- Add drop shadow to keys (No CSS) ✅
- Make the app responsive (No CSS) ✅

5. Include a link to your Azure App in your PR comments.

## Extra Credit

- Add an appropriate sound when the UI letter buttons are clicked (5 points)
- Add creative styling following good UI/UX practices
  - Look at products by Google or some of the vuetify components for good UX/UI ideas
  - https://m2.material.io/design/guidelines-overview

Commit:
#1 mostly done

Only issue is when guess/secret word has multiple of the same letter due to indexOf only grabbing first

Commit:
Merge branch 'Assignment2_TAsk_1' into Assignment2

Commit:
Tests

Commit:
Fixed tests

SergeiAshikhmin committed:
Add gradient to guesses blocks and keys (with CSS)

A-Tanner commented on Assignment2/tests/game.test.ts:
```diff
@@ -0,0 +1,90 @@
+  game.guess.addLetter("T");
+  game.submitGuess();
+
+  expect(updateValidWords(game)).not.toContain("hands");
+  expect(updateValidWords(game)).not.toContain("snare");
+  expect(updateValidWords(game)).not.toContain("traps");
+  expect(updateValidWords(game)).not.toContain("apart");
+  expect(updateValidWords(game)).not.toContain("aptly");
+
+  expect(updateValidWords(game)).toContain("hello");
+  expect(updateValidWords(game)).toContain("holds");
+  expect(updateValidWords(game)).toContain("smell");
+  expect(updateValidWords(game)).toContain("messy");
+  expect(updateValidWords(game)).toContain("elves");
```
Extensive!

A-Tanner replied:
Extensive! While manual tests are sufficient, consider ways to test this logic in more general cases

A-Tanner reviewed (commented):
1. Create a custom component to show available words ✅
- Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ✅
- Create unit tests for the validWords method ✅
- Display the number of valid words based on the entered letters ✅
- When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
- When a word is clicked, the word should be entered as the guess but not submitted ✅
- Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅

2. Convert the sidebar to an App Bar ✅
- Name of the app on the left with an icon ✅
- Choose an icon for the app (from Material Design or somewhere on the web) ✅
- Clicking the name and icon of the app take you to the home page ✅
- Hamburger menu on the right ✅
- Menu option for an About page ✅
  - Create an About page that has a few sentences about this class project and how great Meg is ✅
  - Make sure this has a back button ✅

3. Add a settings dialog ✅
- Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
- Option for dark and light mode ✅
- Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
- The above two items must be implemented with built in Veutify features ✅

4. Style the main game page ✅
- Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
- Add gradient to guesses blocks and keys (with CSS) ✅
- Add drop shadow to keys (No CSS) ✅
- Make the app responsive (No CSS) ✅

5. Include a link to your Azure App in your PR **comments.✅**
## Extra Credit

- Add an appropriate sound when the UI letter buttons are clicked (5 points)
- Add creative styling following good UI/UX practices
  - Look at products by Google or some of the vuetify components for good UX/UI ideas
  - https://m2.material.io/design/guidelines-overview

A-Tanner commented on Assignment2/pages/index.vue:
```diff
@@ -0,0 +1,76 @@
+      </div>
+
+      <WordList
+      v-if="game.gameState === GameState.Playing"
```
Nice attention to detail, it adds a level of polish to the experience

A-Tanner commented on Assignment2/pages/about.vue:
```diff
@@ -0,0 +1,59 @@
+					About Our Teachers
+				</h1>
+				<div class="text-body-1 mb-3">
+					<strong>Grant</strong> and <strong>Mag</strong> are the instructors for this course. They are both very
```
Who is this Mag person? 😅

A-Tanner reviewed (commented):
1. Create a custom component to show available words ✅
- Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ✅
- Create unit tests for the validWords method ✅
- Display the number of valid words based on the entered letters ✅
- When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
- When a word is clicked, the word should be entered as the guess but not submitted ✅
- Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅

2. Convert the sidebar to an App Bar ✅
- Name of the app on the left with an icon ✅
- Choose an icon for the app (from Material Design or somewhere on the web) ✅
- Clicking the name and icon of the app take you to the home page ✅
- Hamburger menu on the right ✅
- Menu option for an About page ✅
  - Create an About page that has a few sentences about this class project and how great Meg is ✅
  - Make sure this has a back button ✅

3. Add a settings dialog ✅
- Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
- Option for dark and light mode ✅
- Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
- The above two items must be implemented with built in Veutify features ✅

4. Style the main game page ✅
- Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
- Add gradient to guesses blocks and keys (with CSS) ✅
- Add drop shadow to keys (No CSS) ✅
- Make the app responsive (No CSS) ✅

5. Include a link to your Azure App in your PR comments.✅
## Extra Credit

- Add an appropriate sound when the UI letter buttons are clicked (5 points)
- Add creative styling following good UI/UX practices
  - Look at products by Google or some of the vuetify components for good UX/UI ideas
  - https://m2.material.io/design/guidelines-overview
  
  Great work you two! Solid implementation and an example of good teamwork. I appreciate the thoughtful commit names you two used to communicate.
